internal/serverinstall/ecs.go

```diff
@@ -0,0 +1,2190 @@
+				{
+					Name:   aws.String("default-for-az"),
+					Values: []*string{aws.String("true")},
+				},
+			},
+		})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		for _, subnet := range desc.Subnets {
+			subnets = append(subnets, subnet.SubnetId)
+		}
+		if len(desc.Subnets) == 0 {
+			return nil, nil, fmt.Errorf("no default subnet information found")
+		}
+		vpcID = desc.Subnets[0].VpcId
+		return subnets, vpcID, nil
+	}
+
+	subnets = make([]*string, len(i.config.subnets))
+	for j := range i.config.subnets {
+		subnets[j] = &i.config.subnets[j]
+	}
+	s.Update("Using provided subnets for Service networking")
+	subnetInfo, err := ec2Svc.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		SubnetIds: subnets,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if len(subnetInfo.Subnets) == 0 {
+		return nil, nil, fmt.Errorf("no subnet information found for provided subnets")
+	}
+
+	vpcID = subnetInfo.Subnets[0].VpcId
+
+	return subnets, vpcID, nil
+}
+
+func registerTaskDefinition(def *ecs.RegisterTaskDefinitionInput, ecsSvc *ecs.ECS) (*ecs.TaskDefinition, error) {
+	// AWS is eventually consistent so even though we probably created the
+	// resources that are referenced by the task definition, it can error out if
+	// we try to reference those resources too quickly. So we're forced to guard
+	// actions which reference other AWS services with loops like this.
+	var taskOut *ecs.RegisterTaskDefinitionOutput
+	var err error
+	for i := 0; i < 30; i++ {
+		taskOut, err = ecsSvc.RegisterTaskDefinition(def)
+		if err == nil {
+			break
+		}
+
+		// if we encounter an unrecoverable error, exit now.
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == "ResourceConflictException" {
+				return nil, err
+			}
+		}
+
+		// otherwise sleep and try again
+		time.Sleep(2 * time.Second)
+	}
+	return taskOut.TaskDefinition, nil
+}
+
+func (i *ECSInstaller) LaunchRunner(
+	ctx context.Context,
+	s LifecycleStatus,
+	log hclog.Logger,
+	ui terminal.UI,
+	sess *session.Session,
+	env []string,
+	executionRoleArn, logGroup, ulid string,
+) (*string, error) {
+
+	s.Status("Installing Waypoint runner into ECS...")
+
+	defaultStreamPrefix := fmt.Sprintf("waypoint-runner-%d", time.Now().Nanosecond())
+	logOptions := buildLoggingOptions(
+		nil,
+		i.config.region,
+		logGroup,
+		defaultStreamPrefix,
+	)
+
+	cpu := defaultTaskCPU
+	mem := defaultTaskMemory
+	grpcPort, _ := strconv.Atoi(defaultGrpcPort)
+
+	envs := []*ecs.KeyValuePair{}
+	for _, line := range env {
+		idx := strings.Index(line, "=")
+		if idx == -1 {
+			// Should never happen but let's not crash.
+			continue
+		}
+
+		key := line[:idx]
+		value := line[idx+1:]
+		envs = append(envs, &ecs.KeyValuePair{
+			Name:  aws.String(key),
+			Value: aws.String(value),
+		})
+	}
+
+	def := ecs.ContainerDefinition{
+		Essential: aws.Bool(true),
+		Command: []*string{
+			aws.String("runner"),
+			aws.String("agent"),
+			aws.String("-vvv"),
+			aws.String("-liveness-tcp-addr=:1234"),
```

Comment:
1234?

Reply:
We discussed this internally, and are going to leave it for now. I borrowed this from the [k8s install](https://github.com/hashicorp/waypoint/blob/afb837e3f2e77ca102926fbddcefa114bdbf4971/internal/serverinstall/k8s.go#L999-L1002)